core-java-modules/core-java-arrays-operations-advanced-2/src/test/java/com/baeldung/movezerototheend/MoveZeroesToTheEndOfAnArrayUnitTest.java

```diff
@@ -0,0 +1,40 @@
+package com.baeldung.movezerototheend;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+public class MoveZeroesToTheEndOfAnArrayUnitTest {
+    static final int[] EXPECTED = new int[] { 1, 2, 3, 4, 0, 0 };
+
+    @Test
+    void whenCreatingANewArrayAndCopyingValues_thenGetTheExpectedResult() {
+        int[] array = new int[] { 1, 2, 0, 3, 4, 0 };
+        int[] result = new int[array.length];
+        int idx = 0;
+        for (int n : array) {
+            if (n != 0) {
+                result[idx++] = n;
+            }
+        }
+        assertArrayEquals(EXPECTED, result);
+    }
+
+    @Test
+    void whenMovingZeroInTheOriginalArray_thenGetTheExpectedResult() {
+        int[] array = new int[] { 1, 2, 0, 3, 4, 0 };
+        int idx = 0;
+        for (int n : array) {
+            if (n != 0) {
+                array[idx++] = n;
+            }
+            System.out.println(Arrays.toString(array));
```

Comment:
Can we remove this? If not, we have a guideline to use SLF4J for logging and not System.out

Reply:
I put it here to print step result, and used the output in the article. Of course, this shouldn't be in the code. I forgot to remove it. Thank you for spotting it. 👍 
fixed in the new commit.